### PR TITLE
Add support for multi-resource statements

### PR DESCRIPTION
**Pull request by Buratti**
Adds support for multiple actions and resources in the same Policy's Statement

### Motivation:

It is common practice to return a statement that describes which integrations are available to the authorized principal. Such statement usually allows the `"execute-api:Invoke"` action on a list of integrations ARN.
The current design of the `APIGatewayLambdaAuthorizerPolicyResponse` didn't allow that and instead forces the user to generate a long list of statements, each allowing access to just one integration.

### Modifications:

`action` and `resource` are now arrays, a new initializer has been added which takes arrays instead of strings. The existing initializer has been changed to automatically initialize a one-element array.
A new test has been added to validate the correct encoding and decoding.

### CODE REVIEW

File: Sources/AWSLambdaEvents/APIGatewayLambdaAuthorizers.swift [hunk 1]
```diff
@@ -81,11 +81,17 @@ public struct APIGatewayLambdaAuthorizerPolicyResponse: Codable {
...
             }
 
-            public let action: String
+            public let action: [String]
             public let effect: Effect
-            public let resource: String
+            public let resource: [String]
 
             public init(action: String, effect: Effect, resource: String) {
+                self.action = [action]
```
Comment (sebsto) [inline Sources/AWSLambdaEvents/APIGatewayLambdaAuthorizers.swift:89]: I would call the other init() method to avoid duplicating code

**Review by sebsto — COMMENTED** [review]
Thank you for having added this. I just suggested one small change

**Comment by Buratti** [comment]
Hi @sebsto I have ran the soundness.sh script but apparently it changed a few files that weren't supposed to be edited in this PR. You want me to revert the last commit?

**Review by sebsto — COMMENTED** [review]
Thank you for the proposed changes.
I would keep only the change in the file [Sources/AWSLambdaEvents/APIGatewayLambdaAuthorizers.swift](https://github.com/swift-server/swift-aws-lambda-events/pull/52/files#diff-3ca2fc2090d938b577cec6deb6b78bf2fda241faeb58370c632bd6fb8aa02b8b)

Can you revert the other changes made by Swift Format ?  I have another PR in progress that takes care of these.

Thanks

**Comment by sebsto** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
@swift-server-bot add to allowlist

**Comment by sebsto** [comment]
@yim-lee can you help and give me permanent authorisation to trigger a test on any PR ?  Thank you

**Comment by sebsto** [comment]
@Buratti Can you also pull and merge the changes from the main branch ? I just merged a PR that has conflicting changes.  Thank you and sorry for the trouble

**Review by sebsto — APPROVED** [review]
lgtm

**Comment by sebsto** [comment]
@Buratti one more little soundness fix

**Comment by Buratti** [comment]
Hi @sebsto! I'm sorry, forgot to run it again. It should be good now!

**Comment by sebsto** [comment]
Merged. Thank you !

**Pull request merged**